internal/exec/stages/files/passwd.go

```diff
@@ -94,18 +97,23 @@ func (s *stage) createPasswd(config types.Config) error {
 	return nil
 }
 
-// createUsers creates the users as described in config.Passwd.Users.
-func (s stage) createUsers(config types.Config) error {
+// ensureUsers ensures that the users will be created
```

Comment:
Maybe `ensureX ensures that X match the state described in config.Passwd.X.` for both `ensureUsers` & `ensureGroups`?